app/src/main/java/com/google/samples/apps/sunflower/data/UnsplashRepository.kt

```diff
@@ -17,29 +17,28 @@
 package com.google.samples.apps.sunflower.data
 
 import com.google.samples.apps.sunflower.api.UnsplashService
-import kotlinx.coroutines.channels.ConflatedBroadcastChannel
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.MutableStateFlow
 import retrofit2.HttpException
 import java.io.IOException
 
 class UnsplashRepository (private val service: UnsplashService) {
 
-    private val searchResult = ConflatedBroadcastChannel<UnsplashSearchResult>()
+    private val searchResult = MutableStateFlow<UnsplashSearchResult>(UnsplashSearchResult.Idle)
 
     suspend fun getSearchResultStream(query: String): Flow<UnsplashSearchResult> {
```

Comment:
From an architecture perspective, it feels weird that this `getSearchResultStream` is doing both a request to fetch data and also returning the flow. I would consider an alternatives here:

1) Does it really need to return a `Flow`? If it's requesting data, maybe it should be a suspend function and simply return `UnsplashSearchResult`

2) If does need to get updates and it needs to be a `Flow`, maybe make the consumer of this class subscribe to the flow all the time and call `requestData` or `loadData` when needed?